src/FAudio_platform_win32.c

```diff
@@ -202,6 +203,128 @@ static DWORD WINAPI FAudio_AudioClientThread(void *user)
 	return 0;
 }
 
+/* Returns the device index for the default device.
+ * Returns 0 on failure. This will prevent the rest of the FAudio code from 
+ * re-ordering the devices. */
+static uint32_t FAudio_DefaultDeviceIndex(IMMDeviceCollection *device_collection)
```

Comment:
This returns 0 on any error, is this intended? Maybe this should instead return it to the caller?

Reply:
I reworked this to return an `HRESULT` and take `defaultDeviceIndex` as parameter here. It doesn't really change much as setting `defaultDeviceIndex` to 0 just does not change the ordering. We just end up with the same error down the line so might as well return the error code here.